### PR TITLE
Train model for decoding tasks like motor imagery

### PR DESCRIPTION
**Pull request by oreHGA**
When running `train_gpt.py` there are two options for training style:
- `CSM_causal` that generates embeddings from input eeg data
- `decoding` that generates predictions from task based data like motor imagery (left eye, right eye, tongue, foot etc)

This PR support for training a model to generate predictions for downstream tasks like motor imagery. 

To run this, see the finetuning section of the readme, download the datasets and then run `finetune.sh`

I ran 10k training steps twice, model accuracy reported were `95.7% and 92.4%`

Screenshot 2024-07-01 at 8 40 22 PM

### CODE REVIEW

**Comment by zeydabadi** [comment]
Hi,
So do you mean that the 4-class classification accuracy is 95.7%?
